### ISSUE TITLE
Corepack Support

### ISSUE THREAD

**Issue by RichiCoder1** [projen/projen#2484]
NPM in recent versions has added support for a field/tool called [corepack](https://nodejs.org/api/corepack.html) which allows you to use specific package manager w/ a specific (or optional) version without having to manually install it yourself.

This is super useful and it'd be awesome if projen supported this.

In theory this would just be automatically adding `"packageManager"` to the `package.json` whenever the manager is set to Yarn or PNPM (maybe with a version boundary adisory message to run `corepack enable` to use this functionality)

**Comment by mrgrain** [comment]
Yes, I've looked at corepack and it seems quite interesting. 

It's super annoying they only support explicit versions and not npm.

**Comment by RichiCoder1** [comment]
Yup. They added support for ranges and then immediately took it out 🥲. NPM is a funny one too since I think it's a chicken & egg kind of thing thing.

**Comment by RichiCoder1** [comment]
For anyone else who stumbles up this, a very easy workaround is (for pnpm, substitute yarn for yarn):
```ts
const pnpmVersion = "7.29.0";

project.addFields({
  packageManager: `pnpm@${pnpmVersion}`,
  engines: {
    pnpm: `^${pnpmVersion}`,
  },
});
```

**Comment by mrgrain** [comment]
I kind of want to re-do the whole package manager thing now. Instead of having multiple properties for versions etc, we should have a generic object of sorts. Corepack could be a setting for pnpm and yarn.


PS: Not sure about `engines` part here.
